promql/engine.go

```diff
@@ -3758,57 +3749,65 @@ func preprocessExprHelper(expr parser.Expr, start, end time.Time) bool {
 		case parser.END:
 			n.Timestamp = makeInt64Pointer(timestamp.FromTime(end))
 		}
-		return n.Timestamp != nil
+		return n.Timestamp != nil, n.Timestamp != nil
 
 	case *parser.AggregateExpr:
+		unwrapParenExpr(&n.Expr)
+		unwrapParenExpr(&n.Param)
 		return preprocessExprHelper(n.Expr, start, end)
 
 	case *parser.BinaryExpr:
-		isInvariant1, isInvariant2 := preprocessExprHelper(n.LHS, start, end), preprocessExprHelper(n.RHS, start, end)
+		isInvariant1, shouldWrap1 := preprocessExprHelper(n.LHS, start, end)
+		isInvariant2, shouldWrap2 := preprocessExprHelper(n.RHS, start, end)
 		if isInvariant1 && isInvariant2 {
-			return true
+			return true, true
 		}
 
-		if isInvariant1 {
+		if shouldWrap1 {
```

Comment:
`shouldWrap1 -> shouldWrapLHS`
`shouldWrap2 -> shouldWrapRHS`
`isInvariant1 -> isInvariantLHS`
`isInvariant2 -> isInvariantRHS`

is better?